projects/fal/src/fal/app.py

```diff
@@ -114,12 +119,20 @@ def collect_routes(self) -> dict[RouteSignature, Callable[..., Any]]:
 
     @asynccontextmanager
     async def lifespan(self, app: FastAPI):
-        await _call_any_fn(self.setup)
+        async def _setup():
+            await _call_any_fn(self.setup)
+            self._ready = True
+
+        task = asyncio.create_task(_setup())
         try:
             yield
         finally:
+            await task
             await _call_any_fn(self.teardown)
 
+    def health(self):
+        return {"ready": self._ready}
+
```

Comment:
We could add runtime stuff that we care for here, like fastapi/pydantic/isolate versions, for example.